iphone/cli/hooks/frameworks.js

```diff
@@ -0,0 +1,841 @@
+	}
+
+	/**
+	 * Does an incremental task run by only scanning changed framework folders
+	 * and removing deleted frameworks from the metadata object
+	 *
+	 * @param {Map.<String, String>} changedFiles Map of changed files and their status (created, changed or deleted)
+	 * @return {Promise}
+	 */
+	doIncrementalTaskRun(changedFiles) {
+		let loaded = this.loadFrameworkMetadata();
+		if (!loaded) {
+			return this.doFullTaskRun();
+		}
+
+		this._frameworks.forEach(frameworkInfo => {
+			if (!fs.existsSync(frameworkInfo.path)) {
+				this.logger.trace(`Framework at ${frameworkInfo.path} deleted, removing metadata`);
+				this._frameworks.delete(frameworkInfo.name);
+				return;
+			}
+
+			// Remove any frameworks from deactivated modules
+			if (!this._frameworkPaths.has(frameworkInfo.path)) {
+				this.logger.trace(`Framework at ${frameworkInfo.path} no longer in search path, removing metadata`);
+				this._frameworks.delete(frameworkInfo.name);
+				return;
+			}
+		});
+
+		let changedFrameworks = new Set();
+		changedFiles.forEach((fileStatus, pathAndFilename) => {
+			if (fileStatus === 'created' || fileStatus === 'changed') {
+				let frameworkPath = pathAndFilename.substring(0, pathAndFilename.indexOf('.framework') + 10);
+				if (!changedFrameworks.has(frameworkPath)) {
+					this.logger.trace(`Framework at ${frameworkPath} changed, regenerating metadata`);
+					changedFrameworks.add(frameworkPath);
+				}
+			}
+		});
+
+		return this.inspectFrameworks(changedFrameworks)
+			.then(this.writeFrameworkMetadata.bind(this));
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	loadResultAndSkip() {
+		let loaded = this.loadFrameworkMetadata();
+		if (!loaded) {
+			return this.doFullTaskRun();
+		}
+
+		return Promise.resolve();
+	}
+
+	/**
+	 * Loads stored metadata from disk and recreates the {@link FrameworkInfo}
+	 * objects
+	 *
+	 * @return {Boolean} True if the metadata was sucessfully loaded, false if not
+	 */
+	loadFrameworkMetadata() {
+		try {
+			let metadata = JSON.parse(fs.readFileSync(this._metadataPathAndFilename));
+			Object.keys(metadata).forEach(frameworkPath => {
+				let frameworkMetadata = metadata[frameworkPath];
+				this._frameworks.set(frameworkMetadata.name, new FrameworkInfo(
+					frameworkMetadata.name,
+					frameworkMetadata.path,
+					frameworkMetadata.type,
+					new Set(frameworkMetadata.architectures)
+				));
+			});
+			return true;
+		} catch (e) {
+			return false;
+		}
+	}
+
+	/**
+	 * Saves the internal matadata object to disk for reuse on subsequent builds
+	 */
+	writeFrameworkMetadata() {
+		let metadataObject = {};
+		this._frameworks.forEach(frameworkInfo => {
+			metadataObject[frameworkInfo.path] = {
+				name: frameworkInfo.name,
+				path: frameworkInfo.path,
+				type: frameworkInfo.type,
+				architectures: Array.from(frameworkInfo.architectures)
+			};
+		});
+		if (!fs.existsSync(this._outputDirectory)) {
+			wrench.mkdirSyncRecursive(this._outputDirectory);
```

Comment:
Don't use `wrench`. Use `fs-extra`'s `fs.mkdirsSync()`.

Reply:
I wanted to, but since `fs-extra` is no direct dependency in our package.json yet i resorted to `wrench`.